### PR TITLE
Enhance flannel discovery by listing DaemonSets across all namespaces

### PR DESCRIPTION
**Pull request by aswinayyolath**
Signed-off-by: Aswin A <aswin6303@gmail.com>

Contributes to:  https://github.com/submariner-io/submariner/issues/3210

### CODE REVIEW

**Comment by aswinayyolath** [comment]
It would be nice if someone can review the code.. This is done as part below linked  issue 

https://github.com/submariner-io/submariner/issues/3204#issuecomment-2454663415

**Comment by yboaron** [comment]
@aswinayyolath Thanks for contributing!

If you want to verify the change locally, you can
*   clone [subctl repo](https://github.com/submariner-io/subctl)
*  update flannel discovery code in subctl directory (vendor/github.com/submariner-io/submariner-operator/pkg/discovery/network/flannel.go )
*  Build subctl binary  - `make cmd/bin/subctl`
*  Use local subctl binary (cmd/bin/subctl ) when you run subctl commands (deploy-broker , join )

**Comment by aswinayyolath** [comment]
ohh Thats great! thanks a lot @yboaron for the info I will give it a go

**Comment by aswinayyolath** [comment]
@yboaron I can't find the mentioned path in this repo [subctl repo](https://github.com/submariner-io/subctl)

## No flannel.go in subctl

```
Aswin 🔥🔥🔥 $ pwd
/tmp/subctl
Aswin 🔥🔥🔥 $ find . -name flannel.go
Aswin 🔥🔥🔥 $
```

### But it's there in `submariner-operator`

```
Aswin 🔥🔥🔥 $
Aswin 🔥🔥🔥 $ find . -name flannel.go
./pkg/discovery/network/flannel.go
Aswin 🔥🔥🔥 $
```

The problem is I am facing issues invoking make targets ...

**Comment by aswinayyolath** [comment]
Okay I found it after downloading all the packages locally using `go mod vendor`

```
Aswin 🔥🔥🔥 $ find . -name "flannel.go"

./vendor/github.com/submariner-io/submariner-operator/pkg/discovery/network/flannel.go
Aswin 🔥🔥🔥 $

```

**Comment by aswinayyolath** [comment]
## Before change

Screenshot 2024-11-09 at 7 32 48 PM

## After Change

Screenshot 2024-11-09 at 7 17 20 PM

The CNI is correctly getting identified as Flannel now

**Comment by aswinayyolath** [comment]
### Running unit tests


<details>
<summary>All unit tests Passed</summary>
<br>

```
[14:21:27.514] [dir=submariner-operator-asw]$ go tool cover -func unit.coverprofile
github.com/submariner-io/submariner-operator/controllers/apply/apply.go:40:					DaemonSet					84.6%
github.com/submariner-io/submariner-operator/controllers/apply/apply.go:100:					Deployment					89.5%
github.com/submariner-io/submariner-operator/controllers/apply/apply.go:145:					ConfigMap					89.5%
github.com/submariner-io/submariner-operator/controllers/apply/apply.go:190:					Service						89.7%
github.com/submariner-io/submariner-operator/controllers/apply/apply.go:257:					awaitResource					100.0%
github.com/submariner-io/submariner-operator/controllers/apply/apply.go:263:					isImmutableError				66.7%
github.com/submariner-io/submariner-operator/controllers/apply/apply.go:277:					copyLabels					100.0%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/cleanup.go:41:			doCleanup					86.4%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/cleanup.go:103:			removeFinalizer					100.0%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/cleanup.go:109:			removeLighthouseConfigFromCustomDNSConfigMap	100.0%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:98:	Reconcile					80.0%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:164:	getServiceDiscovery				100.0%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:175:	addFinalizer					83.3%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:191:	newLighthouseAgent				75.0%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:268:	newLighthouseDNSConfigMap			100.0%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:298:	newCoreDNSCustomConfigMap			100.0%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:307:	newLighthouseCoreDNSDeployment			100.0%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:377:	newLighthouseCoreDNSService			100.0%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:418:	getCustomCoreDNSNamespace			66.7%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:426:	updateDNSCustomConfigMap			94.7%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:465:	configureDNSConfigMap				100.0%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:483:	updateLighthouseConfigInConfigMap		100.0%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:539:	findCoreDNSListeningPort			100.0%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:551:	configureOpenshiftClusterDNSOperator		71.4%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:567:	updateLighthouseConfigInOpenshiftDNSOperator	85.0%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:610:	getUpdatedForwardServers			92.9%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:670:	getImagePath					100.0%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:676:	SetupWithManager				0.0%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:691:	ensureLightHouseAgent				71.4%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:715:	ensureLighthouseCoreDNSDeployment		62.5%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:741:	ensureLighthouseCoreDNSService			75.0%
github.com/submariner-io/submariner-operator/controllers/servicediscovery/servicediscovery_controller.go:761:	buildDomains					100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/broker_controller.go:48:			Reconcile					72.4%
github.com/submariner-io/submariner-operator/controllers/submariner/broker_controller.go:113:			SetupWithManager				0.0%
github.com/submariner-io/submariner-operator/controllers/submariner/cleanup.go:39:				runComponentCleanup				85.0%
github.com/submariner-io/submariner-operator/controllers/submariner/cleanup.go:100:				removeFinalizer					100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/cleanup.go:106:				ensureServiceDiscoveryDeleted			85.7%
github.com/submariner-io/submariner-operator/controllers/submariner/cleanup.go:121:				newDaemonSet					100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/daemonsets.go:32:				updateDaemonSetStatus				83.3%
github.com/submariner-io/submariner-operator/controllers/submariner/daemonsets.go:57:				checkDaemonSetContainers			80.0%
github.com/submariner-io/submariner-operator/controllers/submariner/daemonsets.go:88:				retrieveDaemonSetContainerStatuses		85.7%
github.com/submariner-io/submariner-operator/controllers/submariner/daemonsets.go:104:				findPodsBySelector				75.0%
github.com/submariner-io/submariner-operator/controllers/submariner/gateway_resources.go:52:			newGatewayDaemonSet				100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/gateway_resources.go:82:			newGatewayPodTemplate				77.3%
github.com/submariner-io/submariner-operator/controllers/submariner/gateway_resources.go:253:			reconcileGatewayDaemonSet			100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/gateway_resources.go:275:			buildGatewayStatusAndUpdateMetrics		42.9%
github.com/submariner-io/submariner-operator/controllers/submariner/gateway_resources.go:305:			retrieveGateways				60.0%
github.com/submariner-io/submariner-operator/controllers/submariner/gateway_resources.go:329:			toInt32						100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/globalnet_resources.go:39:			reconcileGlobalnetDaemonSet			80.0%
github.com/submariner-io/submariner-operator/controllers/submariner/globalnet_resources.go:60:			newGlobalnetDaemonSet				100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/loadbalancer_resources.go:48:		reconcileLoadBalancer				81.8%
github.com/submariner-io/submariner-operator/controllers/submariner/loadbalancer_resources.go:74:		getOCPPlatformType				77.8%
github.com/submariner-io/submariner-operator/controllers/submariner/loadbalancer_resources.go:93:		newLoadBalancerService				100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/metrics_proxy_resources.go:38:		reconcileMetricsProxyDaemonSet			100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/metrics_proxy_resources.go:44:		newMetricsProxyDaemonSet			100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/metrics_proxy_resources.go:84:		metricProxyContainer				100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/np_syncer_resources.go:37:			removeNetworkPluginSyncerDeployment		83.3%
github.com/submariner-io/submariner-operator/controllers/submariner/route_agent_resources.go:40:		reconcileRouteagentDaemonSet			100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/route_agent_resources.go:47:		newRouteAgentDaemonSet				72.7%
github.com/submariner-io/submariner-operator/controllers/submariner/servicediscovery_resources.go:34:		serviceDiscoveryReconciler			69.6%
github.com/submariner-io/submariner-operator/controllers/submariner/servicediscovery_resources.go:99:		newServiceDiscoveryCR				100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_controller.go:109:		NewReconciler					100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_controller.go:133:		Reconcile					75.0%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_controller.go:274:		getImagePath					100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_controller.go:279:		getSubmariner					100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_controller.go:290:		addFinalizer					85.7%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_controller.go:307:		SetupWithManager				0.0%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_controller.go:330:		setupSecretSyncer				80.0%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_controller.go:390:		cancelSecretSyncer				50.0%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_controller.go:402:		getBrokerClient					85.7%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_controller.go:427:		getAuthorizedBrokerClientFor			0.0%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_metrics.go:69:			init						100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_metrics.go:73:			recordGateways					100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_metrics.go:81:			recordGatewayCreationTime			0.0%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_metrics.go:88:			recordNoConnections				100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_metrics.go:92:			recordConnection				0.0%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_networkdiscovery.go:31:		getClusterNetwork				23.1%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_networkdiscovery.go:58:		discoverNetwork					100.0%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_networkdiscovery.go:78:		getCIDR						88.9%
github.com/submariner-io/submariner-operator/controllers/submariner/submariner_networkdiscovery.go:101:		getFirstCIDR					66.7%
github.com/submariner-io/submariner-operator/pkg/cidr/cidr.go:51:						unmarshalClusterInfo				87.5%
github.com/submariner-io/submariner-operator/pkg/cidr/cidr.go:67:						ExtractClusterInfo				100.0%
github.com/submariner-io/submariner-operator/pkg/cidr/cidr.go:79:						AddClusterInfoData				88.2%
github.com/submariner-io/submariner-operator/pkg/cidr/cidr.go:112:						IsValid						91.7%
github.com/submariner-io/submariner-operator/pkg/cidr/cidr.go:137:						CheckForOverlappingCIDRs			100.0%
github.com/submariner-io/submariner-operator/pkg/cidr/cidr.go:152:						Allocate					81.8%
github.com/submariner-io/submariner-operator/pkg/cidr/cidr.go:174:						allocateBySize					100.0%
github.com/submariner-io/submariner-operator/pkg/cidr/cidr.go:203:						allocateByCIDR					91.7%
github.com/submariner-io/submariner-operator/pkg/cidr/cidr.go:229:						newAllocationInfo				100.0%
github.com/submariner-io/submariner-operator/pkg/cidr/cidr.go:240:						isOverlappingCIDR				100.0%
github.com/submariner-io/submariner-operator/pkg/cidr/cidr.go:260:						ipToUint					75.0%
github.com/submariner-io/submariner-operator/pkg/cidr/cidr.go:269:						uintToIP					100.0%
github.com/submariner-io/submariner-operator/pkg/cidr/cidr.go:276:						GetValidAllocationSize				0.0%
github.com/submariner-io/submariner-operator/pkg/cidr/cidr.go:299:						nextPowerOf2					0.0%
github.com/submariner-io/submariner-operator/pkg/cidr/cidr.go:311:						IsCIDRPreConfigured				0.0%
github.com/submariner-io/submariner-operator/pkg/crd/updater.go:56:						UpdaterFromRestConfig				0.0%
github.com/submariner-io/submariner-operator/pkg/crd/updater.go:65:						UpdaterFromClientSet				100.0%
github.com/submariner-io/submariner-operator/pkg/crd/updater.go:69:						UpdaterFromControllerClient			0.0%
github.com/submariner-io/submariner-operator/pkg/crd/updater.go:75:						CreateOrUpdateFromEmbedded			80.0%
github.com/submariner-io/submariner-operator/pkg/crd/updater.go:92:						Create						0.0%
github.com/submariner-io/submariner-operator/pkg/crd/updater.go:104:						Update						0.0%
github.com/submariner-io/submariner-operator/pkg/crd/updater.go:116:						Get						0.0%
github.com/submariner-io/submariner-operator/pkg/crd/updater.go:129:						Delete						0.0%
github.com/submariner-io/submariner-operator/pkg/discovery/clustersetip/clustersetip.go:46:			AllocateClustersetIPCIDR			100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/clustersetip/clustersetip.go:50:			ValidateClustersetIPConfiguration		85.7%
github.com/submariner-io/submariner-operator/pkg/discovery/clustersetip/clustersetip.go:76:			GetClustersetIPNetworks				80.0%
github.com/submariner-io/submariner-operator/pkg/discovery/clustersetip/clustersetip.go:104:			assignClustersetIPs				81.0%
github.com/submariner-io/submariner-operator/pkg/discovery/clustersetip/clustersetip.go:148:			ValidateExistingClustersetIPNetworks		88.9%
github.com/submariner-io/submariner-operator/pkg/discovery/clustersetip/clustersetip.go:167:			AllocateCIDRFromConfigMap			78.6%
github.com/submariner-io/submariner-operator/pkg/discovery/clustersetip/config_map.go:45:			CreateConfigMap					71.4%
github.com/submariner-io/submariner-operator/pkg/discovery/clustersetip/config_map.go:62:			NewClustersetIPConfigMap			83.3%
github.com/submariner-io/submariner-operator/pkg/discovery/clustersetip/config_map.go:87:			updateConfigMap					80.0%
github.com/submariner-io/submariner-operator/pkg/discovery/clustersetip/config_map.go:100:			GetConfigMap					100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/clustersetip/config_map.go:106:			DeleteConfigMap					0.0%
github.com/submariner-io/submariner-operator/pkg/discovery/globalnet/config_map.go:44:				CreateConfigMap					71.4%
github.com/submariner-io/submariner-operator/pkg/discovery/globalnet/config_map.go:60:				NewGlobalnetConfigMap				80.0%
github.com/submariner-io/submariner-operator/pkg/discovery/globalnet/config_map.go:97:				updateConfigMap					80.0%
github.com/submariner-io/submariner-operator/pkg/discovery/globalnet/config_map.go:110:				GetConfigMap					100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/globalnet/config_map.go:116:				DeleteConfigMap					0.0%
github.com/submariner-io/submariner-operator/pkg/discovery/globalnet/globalnet.go:46:				AllocateGlobalCIDR				100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/globalnet/globalnet.go:50:				isCIDRPreConfigured				100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/globalnet/globalnet.go:60:				ValidateGlobalnetConfiguration			58.3%
github.com/submariner-io/submariner-operator/pkg/discovery/globalnet/globalnet.go:104:				GetGlobalNetworks				81.2%
github.com/submariner-io/submariner-operator/pkg/discovery/globalnet/globalnet.go:134:				AssignGlobalnetIPs				81.0%
github.com/submariner-io/submariner-operator/pkg/discovery/globalnet/globalnet.go:178:				ValidateExistingGlobalNetworks			88.9%
github.com/submariner-io/submariner-operator/pkg/discovery/globalnet/globalnet.go:197:				AllocateAndUpdateGlobalCIDRConfigMap		78.3%
github.com/submariner-io/submariner-operator/pkg/discovery/network/calico.go:33:				discoverCalicoNetwork				75.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/calico.go:63:				calicoConfigMapExists				87.5%
github.com/submariner-io/submariner-operator/pkg/discovery/network/calico.go:80:				calicoDaemonSetExists				75.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/canal.go:33:					discoverCanalFlannelNetwork			91.7%
github.com/submariner-io/submariner-operator/pkg/discovery/network/flannel.go:35:				discoverFlannelNetwork				95.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/flannel.go:73:				extractCIDRsFromFlannelConfigMap		68.2%
github.com/submariner-io/submariner-operator/pkg/discovery/network/flannel.go:119:				findFlannelConfigMapName			75.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/flannel.go:130:				extractPodCIDRFromNetConfigJSON			71.4%
github.com/submariner-io/submariner-operator/pkg/discovery/network/generic.go:36:				discoverGenericNetwork				85.7%
github.com/submariner-io/submariner-operator/pkg/discovery/network/generic.go:51:				discoverNetwork					85.7%
github.com/submariner-io/submariner-operator/pkg/discovery/network/generic.go:79:				findClusterIPRange				85.7%
github.com/submariner-io/submariner-operator/pkg/discovery/network/generic.go:93:				findClusterIPRangeFromApiserver			100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/generic.go:97:				findClusterIPRangeFromServiceCreation		100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/generic.go:136:				parseServiceCIDRFrom				100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/generic.go:155:				findPodIPRange					100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/generic.go:174:				findPodIPRangeKubeController			100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/generic.go:178:				findPodIPRangeKubeProxy				100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/generic.go:182:				findPodIPRangeFromNodeSpec			80.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/generic.go:193:				parseToPodCidr					100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/kindnet.go:28:				discoverKindNetwork				100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/network.go:42:				Show						88.9%
github.com/submariner-io/submariner-operator/pkg/discovery/network/network.go:60:				Log						0.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/network.go:67:				IsComplete					100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/network.go:71:				Discover					100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/network.go:117:				networkPluginsDiscovery				100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/network.go:128:				getCIDRs					88.9%
github.com/submariner-io/submariner-operator/pkg/discovery/network/openshift4.go:35:				discoverOpenShift4Network			87.5%
github.com/submariner-io/submariner-operator/pkg/discovery/network/openshift4.go:55:				parseOS4Network					76.7%
github.com/submariner-io/submariner-operator/pkg/discovery/network/ovnkubernetes.go:30:				discoverOvnKubernetesNetwork			100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/ovnkubernetes.go:43:				updateClusterNetworkFromConfigMap		100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/pods.go:32:					FindPodCommandParameter				100.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/pods.go:64:					FindPod						80.0%
github.com/submariner-io/submariner-operator/pkg/discovery/network/weavenet.go:28:				discoverWeaveNetwork				100.0%
github.com/submariner-io/submariner-operator/pkg/images/images.go:32:						GetImagePath					0.0%
github.com/submariner-io/submariner-operator/pkg/images/images.go:67:						logIfChanged					0.0%
github.com/submariner-io/submariner-operator/pkg/images/images.go:84:						GetPullPolicy					0.0%
github.com/submariner-io/submariner-operator/pkg/images/images.go:93:						getPullPolicy					0.0%
github.com/submariner-io/submariner-operator/pkg/images/images.go:101:						ParseOperatorImage				100.0%
total:														(statements)					79.1%
```

</details>

File: pkg/discovery/network/flannel.go [hunk 1]
```diff
@@ -28,65 +28,68 @@ import (
...
 )
 
 //nolint:nilnil // Intentional as the purpose is to discover.
 func discoverFlannelNetwork(ctx context.Context, client controllerClient.Client) (*ClusterNetwork, error) {
 	daemonsets := &appsv1.DaemonSetList{}
 
-	err := client.List(ctx, daemonsets, controllerClient.InNamespace(metav1.NamespaceSystem))
+	// Use a label selector to find DaemonSets with the "k8s-app=flannel" label in any namespace
+	labelSelector := controllerClient.MatchingLabels{"k8s-app": "flannel"}
+	err := client.List(ctx, daemonsets, labelSelector)
```
Comment (yboaron) [inline pkg/discovery/network/flannel.go]: Maybe you can use controllerClient.MatchingLabels{"k8s-app": "flannel"} directly in List function

File: pkg/discovery/network/canal.go [hunk 2]
```diff
@@ -43,8 +43,12 @@ func discoverCanalFlannelNetwork(ctx context.Context, client controllerClient.Cl
 		return nil, nil
 	}
 
+	// Get the namespace from the first DaemonSet
+	namespace := daemonsets.Items[0].Namespace
```
Comment (yboaron) [inline pkg/discovery/network/canal.go]: Current canal discovery code search for ds in namespace metav1.NamespaceSystem , you can pass metav1.NamespaceSystem to findFlannelConfigMapName function

**Comment by yboaron** [comment]
> ## Before change
> Screenshot 2024-11-09 at 7 32 48 PM
> ## After Change
> Screenshot 2024-11-09 at 7 17 20 PM
> The CNI is correctly getting identified as Flannel now

Cool! 
Overall code looks good, left 2 comments.

File: pkg/discovery/network/canal.go (hunk 2 above)
Comment (aswinayyolath) [reply pkg/discovery/network/canal.go]: Yeah done

File: pkg/discovery/network/flannel.go (hunk 1 above)
Comment (aswinayyolath) [reply pkg/discovery/network/flannel.go]: Make sense, made the change

File: pkg/discovery/network/flannel_test.go [hunk 3]
```diff
@@ -63,10 +63,13 @@ var flannelDaemonSet = appsv1.DaemonSet{
 	ObjectMeta: metav1.ObjectMeta{
 		Name:      "kube-flannel-ds",
 		Namespace: metav1.NamespaceSystem,
+		Labels:    map[string]string{"k8s-app": "flannel"},
```
Comment (vthapar) [inline pkg/discovery/network/flannel_test.go:92]: Should add a test case for DaemonSet without flannel label returning a generic plugin.

**Review by vthapar — APPROVED** [review]

File: pkg/discovery/network/flannel_test.go (hunk 3 above)
Comment (aswinayyolath) [reply pkg/discovery/network/flannel_test.go:92]: Added the test 

image

**Review by tpantelis — APPROVED** [review]

**Comment by aswinayyolath** [comment]
Thanks for approving the PR. I need to fix below go lint issues as well. I will make required changes!

image

**Review by yboaron — APPROVED** [review]

**Comment by dfarrell07** [comment]
@aswinayyolath thanks for the PR! We think this should have a release note. Would you like to add it? We can show you the process if needed.

**Comment by aswinayyolath** [comment]
Yes please, I will add that. Pls share me the details

**Comment by tpantelis** [comment]
> Yes please, I will add that. Pls share me the details

In the https://github.com/submariner-io/submariner-website project, create a branch off of the **release-notes-0.20** branch. In the _src/content/community/releases/_index.en.md_ file, under the `## v0.20.0` section at the top, add your note under the `### Other changes` section. Then push a PR to the **release-notes-0.20** branch.

**Comment by aswinayyolath** [comment]
Created a PR: https://github.com/submariner-io/submariner-website/pull/1203

**Pull request merged**